src/services/Store/StoreProvider.tsx

```diff
@@ -389,24 +395,41 @@ export const StoreProvider: React.FC = ({ children }) => {
         newRawAccounts[0].wallet,
         newRawAccounts.length
       )(contacts);
-      newRawAccounts.forEach((rawAccount, idx) => {
-        const existingContact = getContactByAddressAndNetworkId(rawAccount.address, networkId);
-        if (existingContact && existingContact.label === translateRaw('NO_LABEL')) {
-          updateContact({
-            ...existingContact,
-            label: newLabels[idx]
-          });
-        } else if (!existingContact) {
-          const newLabel = {
-            label: newLabels[idx],
-            address: rawAccount.address,
-            notes: '',
-            network: rawAccount.networkId,
-            uuid: generateUUID()
-          };
-          createContact(newLabel);
-        }
-      });
+      const contactChanges = newRawAccounts.reduce(
+        (acc, rawAccount, idx) => {
+          const existingContact = getContactByAddressAndNetworkId(rawAccount.address, networkId);
+          if (existingContact && existingContact.label === translateRaw('NO_LABEL')) {
+            return {
+              ...acc,
+              updatedContacts: [
+                ...acc.updatedContacts,
+                {
+                  ...existingContact,
+                  label: newLabels[idx]
+                }
+              ]
+            };
+          } else if (!existingContact) {
+            return {
+              ...acc,
+              newContacts: [
+                ...acc.newContacts,
+                {
+                  label: newLabels[idx],
+                  address: rawAccount.address,
+                  notes: '',
+                  network: rawAccount.networkId,
+                  uuid: generateUUID()
+                }
+              ]
+            };
+          }
+          return acc;
+        },
+        { newContacts: [] as ExtendedContact[], updatedContacts: [] as ExtendedContact[] }
+      );
```

Comment:
This looks like a create or update logic.
It should be the responsibility of the slice.
Move to redux.

FYI: there appears to already be a createOrUpdate defined